server/src/main/java/org/elasticsearch/ReleaseVersions.java

```diff
@@ -104,9 +105,14 @@ private static IntFunction<String> lookupFunction(NavigableMap<Integer, List<Ver
                     // the next version is just a guess - might be a newer revision, might be a newer minor or major...
                     lowerBound = nextVersion(lastItem(lowerRange.getValue())).toString();
                 } else {
-                    // we know about all preceding versions - how can this version be less than everything else we know about???
-                    assert false : "Could not find preceding version for id " + id;
-                    lowerBound = "snapshot[" + id + "]";
+                    // a really old version we don't otherwise know about
+                    // assume it's an old version id
+                    // this will no longer be the case with ES 10 (which won't know about ES v8.x where we introduced separated versions)
+                    // SEP for now
+                    @UpdateForV9
+                    // @UpdateForV10
+                    Version oldVersion = Version.fromId(id);
```

Comment:
This is needed for `DocValueOnlyFieldsIT`, and also probably for error messages trying to form clusters with really old versions